src/main/java/org/jenkinsci/plugins/parallel_test_executor/TargetBranchPreviousBuildFinder.java

```diff
@@ -1,36 +1,27 @@
 package org.jenkinsci.plugins.parallel_test_executor;
 
 import hudson.Extension;
-import hudson.model.ItemGroup;
+import hudson.model.Item;
 import hudson.model.Job;
 import hudson.model.Run;
 import hudson.model.TaskListener;
-import jenkins.branch.Branch;
 import jenkins.scm.api.SCMHead;
 import jenkins.scm.api.mixin.ChangeRequestSCMHead;
-import org.jenkinsci.plugins.workflow.job.WorkflowJob;
-import org.jenkinsci.plugins.workflow.multibranch.WorkflowMultiBranchProject;
 
 /**
- * Returns the latest build from the target branch if the current build is a pull request.
+ * Returns the latest build from the target branch if the current build comes from a change request.
  */
 @Extension(optional = true)
 public class TargetBranchPreviousBuildFinder implements PreviousBuildFinder {
     @Override
     public Run<?, ?> find(Run<?, ?> b, TaskListener listener) {
         Job<?, ?> project = b.getParent();
-        ItemGroup itemGroup = project.getParent();
-        if (itemGroup instanceof WorkflowMultiBranchProject) {
-            WorkflowMultiBranchProject multiBranchProject = (WorkflowMultiBranchProject) itemGroup;
-            Branch branch = multiBranchProject.getProjectFactory().getBranch((WorkflowJob) project);
-            SCMHead head = branch.getHead();
-            if (head instanceof ChangeRequestSCMHead) {
-                SCMHead target = ((ChangeRequestSCMHead) head).getTarget();
-                String targetName = target.getName();
-                WorkflowJob targetBranch = multiBranchProject.getItemByBranchName(targetName);
-                if (targetBranch != null) {
-                    return targetBranch.getLastBuild();
-                }
+        SCMHead head = SCMHead.HeadByItem.findHead(project);
+        if (head instanceof ChangeRequestSCMHead) {
+            SCMHead target = ((ChangeRequestSCMHead) head).getTarget();
+            Item targetBranch = project.getParent().getItem(target.getName());
```

Comment:
Presumes that the target name contains no metacharacters. Will work for the typical cases like `master` or `main`; will not work for e.g. `integrate/prod`.

Reply:
Actually, it works as expected. Both the scm head and item name uses the url-escaped form.